### PR TITLE
closes #63 closes #66 closes #71 interp vignette

### LINKED ISSUE #63: ADVS Vignette: explain interpolation
See https://github.com/pharmaverse/admiralpeds/issues/62

### LINKED ISSUE #66: General Issue: Ben's sporadic review
### Background Information

I don't understand why admiral is in parentheses?  It looks confusing to me?  Is this how it is done in other extensions? 
image

Make this a link - it will never change 
image

What about a note on future templates that are coming and asking commmunity to fill out an issue if they think one might be handy 
image

These roxygen comments seem to have come through - I think you can do `#' #` 
image
 
 Can these be made into regular quotes `"` this has caused us issues with CRAN 
 
image

What is updated
image

### Definition of Done

_No response_

### LINKED ISSUE #71: derive_params_growth_height() should only use WHO weight-for-length and this derivation only relevant for up to 2 years of age
David review comment suggestion:

We should remove the WHO weight-for-height charts from our package and only use WHO weight-for-length for <2 years of age and  clarify nobody should ever use this derivation for >=2 years (as they would really use BMI-for-age).

Also please double check this function has no adding/subtracting of 0.7 cm. I don't think it does, I think in David's example he was just using the WHO weight-for-height metadata instead of weight-for-length that is suggested for <=2 years.

Suggested vignette wording from David:

_Parameters for Weight by Height z-score and percentile for children under 2 years of age could be achieved using the following code, assuming the "HEIGHT" parameter is always collected as recumbent length. You'll notice the `derive_params_growth_height()` function used is very similar to that used above, but now passing in height instead of age.
We consulted with an acting pediatrician who advised that they are rarely guided as to which measure is in the data. It depends mainly on the age at which the child can stand (2 years on average) as well as the preference of the measurer. Since our package template and example function calls below calculate this parameter only for children under 2 years of age, we have assumed that the values in the "HEIGHT" parameter data are for recumbent length.  If you have strict CRF (case report form) guidelines that standing height (rather than length) was measured, you could add 0.7 cm to these values to approximate the child’s recumbent length._

### PR DESCRIPTION
**Pull request by rossfarrugia**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -113,9 +123,18 @@ bmi_for_age <- who_bmi_for_age_boys %>%
...
+    ) %>%
+    # Interpolate the AGE by SEX
+    derive_interp_records(
+      by_vars = exprs(SEX),
+      parameter = "BMI"
+    ) %>%
+    # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old which
+    # must come from WHO metadata only
+    filter(AGE >= 730.5 & AGE <= 7305)) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd:137]: In the comment line 137, add the brackets for the function e.g. derive_params_growth_age()

Comment (Fanny-Gautier) [reply vignettes/advs.Rmd:137]: Cosmetic update: In the displayed data, is it possible to order the variables as following: AGE, AGEU, SEX, L, M, S, P95, Sigma ?

File: vignettes/advs.Rmd [hunk 2]
```diff
@@ -143,12 +162,18 @@ the WHO standards, as no equivalent for \>2 years of age are provided by CDC.
 
 ### Reference Files for Weight by Length/Height
 
-WHO provides additional standards for weight by body length and weight
-by height, instead of age. For these we again combined the metadata for
-sex so that we have a single reference file by body length and a
-single reference file by height. We also give the values and units
-consistent variable names so that we can pass either metadata into the
-same function. We do this using the following code:
+WHO provides additional standards for weight by body length (recumbent length),
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd:165]: Line 159 "The above example only shows the parameter BMI, but similar follows for the
other parameters like weight and height.". Should we mention here that there is no `Sigma` nor `P95` variables for Height and Weight ?

File: vignettes/advs.Rmd [hunk 3]
```diff
@@ -486,9 +448,23 @@ advs <- advs %>%
...
+We consulted with an acting pediatrician who advised that rarely investigators
+are guided as to which measure to use. It depends mainly on the age
+at which the child is able to stand (2 years on average) as well
+as the preference of the measurer.
+
+If you have strict CRF (case report form) guidelines that standing height
+(rather than body length) was measured, you could add 0.7 cm to these values to
+approximate the child’s body length. Or alternatively you could provide a
+different source of reference files by height which could allow for this
+parameter derivation for children >2 years of age, but this is not our package
```
Comment (zdz2101) [inline vignettes/advs.Rmd:465]:
```suggestion
parameter derivation for children \>2 years of age, but this is not our package
```

super minor to keep consistent with usage of operators above

**Review by zdz2101 — COMMENTED** [review]
Looks great! Super small/minor aesthetic things

File: vignettes/advs.Rmd (hunk 1 above)
Comment (rossfarrugia) [reply vignettes/advs.Rmd]: sure, updated both

File: vignettes/advs.Rmd (hunk 2 above)
Comment (rossfarrugia) [reply vignettes/advs.Rmd:168]: added

**Comment by Fanny-Gautier** [comment]
I'll review it again tomorrow as there are further (minor?) updates which might be needed due to the updates in #73.

File: vignettes/advs.Rmd [hunk 4]
```diff
@@ -113,8 +123,17 @@ bmi_for_age <- who_bmi_for_age_boys %>%
         SEX == 2 ~ "F",
         TRUE ~ NA_character_
       ),
-      AGE = AGE * 30.4375
-    )) %>%
+      # Ensure first that Age unit is "DAYS"
+      AGE = round(AGE * 30.4375)
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd:127]: AGEU must be added here (line117) instead of last step. as per updates in #73 , but also after line 118

File: vignettes/advs.Rmd [hunk 5]
```diff
@@ -113,8 +123,17 @@ bmi_for_age <- who_bmi_for_age_boys %>%
...
+    # Interpolate the AGE by SEX
+    derive_interp_records(
+      by_vars = exprs(SEX),
+      parameter = "BMI"
+    ) %>%
+    # Keep patients >= 2 yrs till 20 yrs - Remove duplicates for 730 Days old which
+    # must come from WHO metadata only
+    filter(AGE >= 730.5 & AGE <= 7305)) %>%
   # AGEU is added in metadata, required for derive_params_growth_age()
   mutate(AGEU = "DAYS") %>%
```
Comment (Fanny-Gautier) [inline vignettes/advs.Rmd:138]: AGEU to be move up as per previous comment

File: vignettes/advs.Rmd (hunk 4 above)
Comment (rossfarrugia) [reply vignettes/advs.Rmd:127]: @Fanny-Gautier please can you make these vignette updates in your branch that is working on issue 73 specifically, as its hard to follow including that in this PR which is already doing so much. also that way i can review all those related changes in one PR together as i dont yet really understand why moving the AGEU bit is needed here. will be easier if i can review all together. i'm going to go ahead and merge this PR to main as all previous comments are resolved and then we can merge main into the other open PRs so all feature branches have all these latest changes available.
